parlai/scripts/train_model.py

```diff
@@ -606,13 +620,45 @@ def validate(self):
             ):
                 logging.info('task solved! stopping.')
                 return True
+        elif (
+                self.save_top_k > 1
+                and self.opt.get('model_file')
+                and (
+                    len(self.best_k_models) < self.save_top_k
+                    or self.valid_optim * new_valid > self.valid_optim * self.best_k_models[-1][1]
+                    # if new validation metric is better than kth saved model metric
+                )
+            ):
+            self.impatience = 0
+            model_rank = sum(new_valid < saved_model_prop[1] for saved_model_prop in self.best_k_models)
+            model_suffix = '_' + ordinal(model_rank+1) + '.' + str(self._train_steps)
+            self.best_k_models.insert(model_rank, [self.opt['model_file']+model_suffix, new_valid])
+            self.save_model(model_suffix) # Save model as "model_nth.<number_of_train_steps>"
+            self.saved = True
+            self._modify_next_rank_checkpoints(model_rank)
+            if (
+                opt['validation_metric_mode'] == 'max'
+                and self.best_k_models[-1][1] >= opt['validation_cutoff']
+            ) or (
+                opt['validation_metric_mode'] == 'min'
+                and self.best_k_models[-1][1] <= opt['validation_cutoff']
+            ):
+                logging.info('task solved! stopping.')
+                return True
```

Comment:
i think this check requires that we look at `self.best_k_models[0]`, right? since we're looking at the best metric

Reply:
My idea here was that we want the last saved model's metric to be better than `validation_cutoff` so I checked the last one with `self.best_k_models[-1]`